core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1899,6 +1899,54 @@ object Stream {
     def foldMonoid(implicit O: Monoid[O]): Stream[F, O] =
       self.fold(O.empty)(O.combine)
 
+    def groupWithin(n: Int, d: FiniteDuration)(
+        implicit F: Effect[F],
+        timer: Timer[F],
+        ec: ExecutionContext): Stream[F, Segment[O, Unit]] = {
+      type Tick = Int
+
+      Stream.eval(async.unboundedQueue[F, Option[Either[Tick, O]]]).flatMap { q =>
```

Comment:
If I understand this correctly, this `unboundedQueue`, coupled with your `concurrently` producer later on I think leads to unbounded space usage. You might be able to recover some of your bounded behavior by using something like `synchronousNoneTerminated` or `bounded`, but I'm not sure. I think you run into problems with interrupting a pull in those cases. In particular if you have a really fast upstream stream, I'm not sure whether it's possible for `startTimeout`s `enqueue1` attempt to consistently get beat out.